### PR TITLE
Flag controlled RLIMIT_NOFILE for kubelet.

### PR DESCRIPTION
**Pull request by dchen1107**
Default: 1000000

During some tests, I observed some error messages logged by cAdvisor:

```
I0917 14:58:50.355545   17206 container.go:369] Failed to update stats for container "/system/system.slice/docker-1d2119f0415897dca023510fb88ba5fc6c894f9f58a0c773c97d15a38f3b9ffb.scope": du command failed on /var/lib/docker/aufs/diff/1d2119f0415897dca023510fb88ba5fc6c894f9f58a0c773c97d15a38f3b9ffb with output  - fork/exec /usr/bin/du: cannot allocate memory, continuing to push stats
```

Above message is actually from golang library, and the message is misleading. The resource having issue here is not memory, instead of NOFILE.

### CODE REVIEW

**Comment by dchen1107** [comment]
cc/ @vishh

File: cmd/kubelet/app/server.go [hunk 1]
```diff
@@ -275,6 +277,7 @@ func (s *KubeletServer) AddFlags(fs *pflag.FlagSet) {
 	fs.BoolVar(&s.ReallyCrashForTesting, "really-crash-for-testing", s.ReallyCrashForTesting, "If true, when panics occur crash. Intended for testing.")
 	fs.Float64Var(&s.ChaosChance, "chaos-chance", s.ChaosChance, "If > 0.0, introduce random client errors and latency. Intended for testing. [default=0.0]")
 	fs.BoolVar(&s.Containerized, "containerized", s.Containerized, "Experimental support for running kubelet in a container.  Intended for testing. [default=false]")
+	fs.Uint64Var(&s.MaxOpenFiles, "max-open-files", 1000000, "Number of files that can be opened by Kubelet process. [default=1000000]")
```
Comment (vishh) [inline cmd/kubelet/app/server.go]: This is fd's right? Not just files, but sockets, dentries, etc?

File: pkg/util/resource_container_linux.go [hunk 2]
```diff
@@ -39,3 +40,7 @@ func RunInResourceContainer(containerName string) error {
 
 	return manager.Apply(os.Getpid())
 }
+
+func ApplyRLimit(maxOpenFiles uint64) {
```
Comment (vishh) [inline pkg/util/resource_container_linux.go]: nit: `ApplyRlimitForSelf`

File: cmd/kubelet/app/server.go (hunk 1 above)
Comment (vishh) [inline cmd/kubelet/app/server.go]: Do we have to expose a flag? Do we expect this number to vary based on the some variable like number of pods, etc?

Comment (dchen1107) [inline cmd/kubelet/app/server.go]: Yes, it is, the name of MaxOpenFiles matches /proc/pid/limits 

# cat /proc/2856/limits 
Limit                     Soft Limit           Hard Limit           Units     
Max cpu time              unlimited            unlimited            seconds   
Max file size             unlimited            unlimited            bytes     
Max data size             unlimited            unlimited            bytes     
Max stack size            8388608              unlimited            bytes     
Max core file size        0                    unlimited            bytes     
Max resident set          unlimited            unlimited            bytes     
Max processes             14829                14829                processes 
Max open files            1000000              1000000              files     
Max locked memory         65536                65536                bytes     
Max address space         unlimited            unlimited            bytes     
Max file locks            unlimited            unlimited            locks     
Max pending signals       14829                14829                signals   
Max msgqueue size         819200               819200               bytes     
Max nice priority         0                    0                    
Max realtime priority     0                    0                    
Max realtime timeout      unlimited            unlimited            us

File: pkg/util/resource_container_linux.go (hunk 2 above)
Comment (dchen1107) [inline pkg/util/resource_container_linux.go]: Done

File: cmd/kubelet/app/server.go (hunk 1 above)
Comment (vishh) [inline cmd/kubelet/app/server.go]: Thanks for clarifying Dawn! LGTM

**Comment by vishh** [comment]
LGTM.

**Comment by erictune** [comment]
Sounds like this will help scalability test

**Comment by derekwaynecarr** [comment]
I think these were many of the same error messages I ran into when trying
to soak a node with a number of best effort pods.  Thanks for the fix!

**Comment by sttts** [comment]
This broke contrib/mesos, fix in #14483.

**Pull request merged**